### DISCUSSION TITLE
error: paste subprocess was killed by signal (Broken pipe)

### DISCUSSION THREAD

**Discussion by jackobtone** [dtcooper/raspotify#594] [Q&A] [answered]
Hello,
I`m trying to install Raspotify on MoodeAudio 8.2.2 installed on Raspberry CM4 + CM4IO Board + JCat Femto card.
I keep on getting such an error while pasting in the installation code in Putty.

dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 /var/cache/apt/archives/raspotify_0.43.10~librespot.v0.4.2-e91bc4b_arm64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)

Any ideas?

**Comment by JasonLG1979** [comment, marked as answer]
Why would you do that? Moode already has a version of librespot installed and it's a purpose built appliance type distro. You're not meant to install things in Moode. You're most likely getting that error because the raspotify package wants to install a binary that's already present (librespot).

**Comment by jackobtone** [comment]
That makes sense, thank You for the reply. I wanted to do it, because I had raspotify in the past, and it worked far better than preinstalled Spotify in Moode in terms of stabillity. Then Ive switched form RPi4 to CM4, played preinstalled Spotify for some time and for example right now I have to restart Spotify renderer in Moode every single time I want to connect any device. It makes me sick, and thats the reason why I`ve tried to install raspotify.

**Reply by JasonLG1979** [reply]
Audio hot plugging isn't supported by librespot at all. It will always crash librespot. That's the equivalent of unplugging a hard drive in the middle of a file transfer. The reason that it works at all even kinda is that the Raspotify service auto restarts when librespot crashes.

**Reply by JasonLG1979** [reply]
They can work around it a bit but at the end of the day it's not up to librespot to monitor the audio outputs of a device. In general though IMHO hot plugging a bunch of DACs is not really the intented use case of Moode either. It's meant to basically create an appliance where once you get everything set up the way you want it nothing ever really changes much.

**Reply by jackobtone** [reply]
Thank You for all of the rpelies. I think, that my problem is solved. Will give some other OS for RPi a try.
